playwright.config.ts

```diff
@@ -1,12 +1,32 @@
 import { PlaywrightTestConfig } from '@playwright/test'
 
 const config: PlaywrightTestConfig = {
+  // The directory where the tests are located
+  // The order of the tests is determined by the file names alphabetically.
   testDir: './e2e-tests',
+
+  // The number of retries before marking a test as failed.
   maxFailures: 1,
-  workers: 1, // NOTE: must be 1 for now, otherwise tests will fail.
+
+  // The number of Logseq instances to run in parallel.
+  // NOTE: must be 1 for now, otherwise tests will fail.
+  workers: 1,
+
+  // 'github' for GitHub Actions CI to generate annotations, plus a concise 'dot'.
+  // default 'list' when running locally.
+  reporter: process.env.CI ? 'github' : 'list',
+
+  // Retry on CI only.
+  // This will reduce the risk of false negatives when running on CI.
+  retries: process.env.CI ? 2 : 0,
```

Comment:
It's always better to fail early than to waste time retrying when tests fail. This is especially important because when e2e tests fail, the editor state can become unpredictable. Therefore, it's best to catch and address the issue at the earliest possible stage to avoid any unnecessary complications.

Reply:
@andelf that's a fair point. The optimal scenario is that tests should be completely independent of each other; in that case, retrying the test won't be an issue since the app state is independent between each test.

I do agree that It's best to address the flakiness in code instead of depending on the retries, though. I will remove it.